test-framework/keycloak-server/src/main/java/io/quarkus/test/keycloak/client/KeycloakTestClient.java

```diff
@@ -375,4 +438,51 @@ private static String urlEncode(String value) {
             throw new RuntimeException(ex);
         }
     }
+
+    private RequestSpecification getSpec() {
+        RequestSpecification spec = RestAssured.given();
+        if (tls != null) {
+            spec = spec.keyStore(tls.keystore(), tls.keystorePassword())
+                    .trustStore(tls.truststore(), tls.truststorePassword());
+        } else {
+            spec = spec.relaxedHTTPSValidation();
+        }
+        return spec;
+    }
+
+    public static class Tls {
```

Comment:
Make it a `record`?

Reply:
Yeah, I tried, but then in the test I have to write `new Tls("client-keystore.p12", "password", "client-truststore.p12", "password")` twice in this particular case, so I thought, given that `KeycloakTestClient` offers a lot of overloads to support defaulting for various properties, I'd use the bean.
What do you think or did I miss a trick to have 2 different constructors for `record` ? I did search for it :-)

Reply:
@gastaldi so now I can write only `new KeycloakTestClient(new Tls())` which works for the convention where client keystore is named `client-keystore.p12`, etc

Reply:
But, I retained record-style getters after converting from `record` to `bean` :-)

Reply:
It would translate to this AFAIK: 

```java
    public record Tls (String keystore, String keystorePassword,
                       String truststore, String truststorePassword) {
        public Tls() {
            this("client-keystore.p12", "password", "client-truststore.p12", "password");
        }
    }
```

Reply:
@gastaldi Cool, I saw this form but thought it was more abut the validation, let me do it quickly